### PR TITLE
feat(Popover): MVP version of Popover component.

### PR DESCRIPTION
**Pull request by juanfabrega**
Ready for review.

### CODE REVIEW

File: src/components/Popover/Popover.module.scss [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/* stylelint-disable */
+.popover {
+  box-shadow:
```
Comment (nathanyoung) [inline src/components/Popover/Popover.module.scss:3]: There's a token for this!

File: src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+const Popover: FC<PopoverProps> = ({
+  arrowColor = undefined,
+  children,
+  content,
+  contentContainerProps = { ...contentContainerDefaults },
+  isOpen,
+  onClickOutside = undefined,
+  placement,
+  portalTarget = document.body,
+  trapFocus = false,
```
Comment (nathanyoung) [inline src/components/Popover/Popover.tsx:83]: Should this be true by default?

File: src/components/Popover/Popover.module.scss (hunk 1 above)
Comment (juanfabrega) [reply src/components/Popover/Popover.module.scss:13]: will address before merging.

File: src/components/Popover/Popover.tsx (hunk 2 above)
Comment (juanfabrega) [reply src/components/Popover/Popover.tsx:103]: Hey @nathanyoung having the `trapFocus = true` by default is not a problem but it does create a big pain for anybody using our components and running tests. The tests fail to run properly and require an `await` placed strategically, which even after a dev gets it right (ends up delaying tests substantially).

Comment (nathanyoung) [reply src/components/Popover/Popover.tsx:103]: ok

**Review by nathanyoung — APPROVED** [review]

**Pull request merged**